src/crypto/index.js

```diff
@@ -973,6 +965,25 @@ Crypto.prototype._getE2eRooms = function() {
     });
 };
 
+
+Crypto.prototype._onToDeviceEvent = function(event) {
+    try {
+        if (event.getType() == "m.room_key"
+            || event.getType() == "m.forwarded_room_key") {
+            this._onRoomKeyEvent(event);
+        } else if (event.getType() == "m.room_key_request") {
+            this._onRoomKeyRequestEvent(event);
+        } else if (event.isBeingDecrypted()) {
```

Comment:
It feels like the ordering is important here? `_onToDeviceEvent` is being called with `m.room.encrypted` events? And only after decryption do we know if they are `m.room_key` etc requests and that's why we recursively call this after decryption?